products/opensuse/main.pm

```diff
@@ -565,6 +565,8 @@ sub load_yast2ui_tests() {
 }
 
 sub load_extra_tests () {
+    return unless (!get_var("INSTALLONLY") && get_var("DESKTOP") !~ /textmode/ && !get_var("DUALBOOT") && !get_var("RESCUECD"));
+
```

Comment:
please factor out the common parts between the two and create a new function for it